src/rulesets-new/json-api-rules/compound-document-rules.ts

```diff
@@ -0,0 +1,27 @@
+import { ResponseBodyRule } from "@useoptic/rulesets-base";
+import { isOpenApiPath } from "../utils";
+
+export const compoundDocuments = new ResponseBodyRule({
+  name: "disallow compound documents",
+  matches: (responseBody, rulesContext) =>
+    !isOpenApiPath(rulesContext.operation.path) &&
+    ["200", "201"].includes(responseBody.statusCode) &&
+    responseBody.contentType === "application/vnd.api+json",
+  rule: (responseAssertions) => {
+    responseAssertions.body.added.not.matches({
```

Comment:
There's `.not` inverse conditions on all our standard matchers - this should provide the inverse of the matched block.